tower-spawn-ready/src/future.rs

```diff
@@ -75,7 +72,9 @@ where
             return Poll::Ready(());
         }
 
-        let result = ready!(this.ready.poll(cx));
+        let result = ready!(this.service.as_mut().expect("illegal state").poll_ready(cx))
```

Comment:
this is uh very unfortunate..........

Reply:
That `expect` (like the ones below) is for `poll` being called again after `Poll::Ready` has been returned. If it is hit, the other `expect`s under would also be hit.